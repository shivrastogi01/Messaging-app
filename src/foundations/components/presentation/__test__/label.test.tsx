import React from "react";
import {
  Label,
  LabelVariants,
  LabelSizes
} from "foundations/components/presentation";
import renderWithTheme from "main/testing/render-with-theme";

describe("Label", () => {
  test("default variant matches the snapshot", () => {
    const { container } = renderWithTheme(<Label>Label</Label>);
    expect(container.firstChild).toMatchSnapshot();
  });

  test("inverse variant matches the snapshot", () => {
    const { container } = renderWithTheme(
      <Label variant={LabelVariants.INVERSE}>Label</Label>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test("active variant matches the snapshot", () => {
    const { container } = renderWithTheme(
      <Label variant={LabelVariants.ACTIVE}>Label</Label>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test("small size matches the snapshot", () => {
    const { container } = renderWithTheme(
      <Label size={LabelSizes.SMALL}>Label</Label>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
