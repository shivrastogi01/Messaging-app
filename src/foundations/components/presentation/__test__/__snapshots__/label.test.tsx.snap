// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label active variant matches the snapshot 1`] = `
.c0 {
  color: #9B9B9B;
  font-family: inherit;
  font-size: inherit;
  font-weight: inherit;
  color: #8E2DE2;
}

<span
  class="c0"
>
  Label
</span>
`;

exports[`Label default variant matches the snapshot 1`] = `
.c0 {
  color: #9B9B9B;
  font-family: inherit;
  font-size: inherit;
  font-weight: inherit;
}

<span
  class="c0"
>
  Label
</span>
`;

exports[`Label inverse variant matches the snapshot 1`] = `
.c0 {
  color: #9B9B9B;
  font-family: inherit;
  font-size: inherit;
  font-weight: inherit;
  color: #FFFFFF;
}

<span
  class="c0"
>
  Label
</span>
`;

exports[`Label small size matches the snapshot 1`] = `
.c0 {
  color: #9B9B9B;
  font-family: inherit;
  font-size: inherit;
  font-weight: inherit;
  font-size: 11px;
}

<span
  class="c0"
>
  Label
</span>
`;
